await-wrap.js

```diff
@@ -30,7 +30,7 @@ class AwaitWrap {
         const fn = model[name];
         // Here we basically assume that unless its already an async function
         //  its callback based and we thenify it
-        if (toString.call(fn).toLowerCase().slice(8, -1) !== 'asyncfunction') {
+        if (!isAsyncFn(fn) && fn.length >= 2) {
```

Comment:
Just codifying our offline discussion.

I think this is fine as a first stake in the ground. If we find that we/others need something differently we can build  allow-/block-list type functionality here to be able to work around this limitation.